content/post/2019-12-19-mutable-api.markdown

```diff
@@ -215,47 +253,36 @@ xml2::xml_replace(xml2::xml_find_all(xml, "//softbreak"),
 
 That code changes nodes in the `xml` object without our assigning it back to it.
 
-The reasons `xml2` works this way is that the package is a binding to [the C libxml2 API](http://xmlsoft.org/), where you are supposed to manage memory allocation manually. [`xml2` does handle memory management for you](https://xml2.r-lib.org/#compared-to-the-xml-package) but [`xml2` docs explain how one should be careful when removing nodes because of possible memory issues](https://xml2.r-lib.org/articles/modification.html#removing-nodes). 
-
-Maybe XML data in itself is unusual for you, and maybe the behaviour above is even more unusual for you, but it's a handy package, [if only to tinker with READMEs](/2019/12/03/readmes/#other-size-indicators). :grin:
+The reasons `xml2` works this way is that the package is a binding to [the C libxml2 API](http://xmlsoft.org/), and the XML objects in the libxml2 C API _are_ mutable. [`xml2` does handle memory management for you](https://xml2.r-lib.org/#compared-to-the-xml-package). 
 
 ## Interfacing an external process that's actually mutable in ps::ps_handle()
 
 Now, speaking of objects that are actually mutable, the [`ps` package](http://ps.r-lib.org/) offers an interesting example: the [`ps_handle()`](http://ps.r-lib.org/reference/ps_handle.html) function creates an object that's essentially a pointer to a system process. System processes are of course mutable, they run, then die, can be suspended, etc.
 
-In the example below we launch a system call using `processx`, create a `ps_handle` object corresponding to it i.e. just an external pointer with an S3 class, and we query its status using `ps`. 
+In the example below we launch a process using `processx`, create a `ps_handle` object corresponding to it i.e. just an external pointer with an S3 class, and we query its status using `ps`. `ps` can't _create_ processes but it can query, list and manipulate them.
 
 
 ```r
-p <- processx::process$new("sleep", "5")
+p <- processx::process$new("sleep", "0.5")
 ```
 
 With such a definition, after 5 seconds the process will no longer exist. 
 
 
 ```r
-p$get_pid()
-```
-
-```
-## [1] 12139
-```
-
-```r
-phandle <- ps::ps_handle(p$get_pid())
-class(phandle)
+(p$get_pid())
 ```
 
 ```
-## [1] "ps_handle"
+## [1] 19022
 ```
 
 ```r
-phandle
+(phandle <- p$as_ps_handle())
```

Comment:
again, outer parens?

Reply:
same reason, is that a bad idea?

Reply:
Yeah, you need it here, but I would just do 
```
phandle <- p$as_ps_handle()
phandle
```
this is less magical

Reply:
but here they're needed to print phandle?